scheduler/cups-driverd.cxx

```diff
@@ -2914,3 +2927,74 @@ regex_string(const char *s)		/* I - String to compare */
 
   return (NULL);
 }
+
+void *
+cupsFileGetsCall(void *input)
+{
+  retval = cupsFileGets(((struct args*)input)->fp,((struct args*)input)->buf,((struct args*)input)->buflen);
+  return NULL;
+}
+
+/*
+  * Alternate to cupsFileGets() 
+  */
+
+char * 
+cupsFileGetsT(cups_file_t *fp,		/* I - CUPS file */
+             char        *buf,		/* O - String buffer */
+	     size_t      buflen)        /* I - Size of string buffer */
+{
+  
+  int flag = 0;
+
+ /*
+  * Creating a new thread ID.
+  */
+
+  pthread_t ptid;
+  struct timeval t1, t2;
+
+  struct args *arg = (struct args *)malloc(sizeof(struct args));
```

Comment:
@pranjanpr if you define the struct via typedef, you can use the new type you created.